pygments/lexers/wgsl.py

```diff
@@ -0,0 +1,415 @@
+"""
+    pygments.lexers.wgsl
+    ~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the WebGPU Shading Language.
+
+    :copyright: Copyright 2006-2023 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, include, bygroups, words, default
+from pygments.token import Comment, Operator, Keyword, Name, \
+    Number, Punctuation, Whitespace
+from pygments import unistring as uni
+
+__all__ = ['WgslLexer']
+
+LF = '\\u000a'
+VT = '\\u000b'
+FF = '\\u000c'
+CR = '\\u000d'
+NextLine = '\\u0085'
+LineSep = '\\u2028'
+ParaSep = '\\u2029'
+LineEndCodePoints = [LF,VT,FF,CR,NextLine,LineSep,ParaSep]
+NotLineEndRE = '[^' + "".join(LineEndCodePoints) + ']'
+LineEndRE = '[' + "".join(LineEndCodePoints) + ']'
+
+# https://www.w3.org/TR/WGSL/#syntax-ident_pattern_token
+ident_pattern_token = '([_{}][{}]+)|[{}]'.format(uni.xid_start,uni.xid_continue,uni.xid_start)
+
+def qw(s):
+    """Returns the non-whitespace tokens in a possibly multiline string.
+    Emulate Perl's qw syntax.
+    """
+    result = []
+    for line in s.splitlines():
+      for word in line.split(' '):
+        if len(word) > 0:
+          result.append(word)
+    return result
```

Comment:
> What's the rationale for using this over a list of words directly? 

It's much easier to copy-paste from the spec or another source, without having to worry about details of getting the quoting just right.  Working with qw is really fluent.

> Also, this could be simplified to use `yield word` instead of appending to a result, which shouldn't matter for the `words` function

I am but a simple Python programmer.  :-)

Reply:
Indeed!  I misread the docs. Will fix.

Reply:
Fixed this. I've removed the qw function. Thanks!